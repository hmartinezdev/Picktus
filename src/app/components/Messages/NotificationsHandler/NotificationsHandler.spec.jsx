import { shallow } from 'enzyme';
import React from 'react';
import NotificationHandler from './NotificationsHandler';

let wrapper;

const setup = (props = {}) => shallow(<NotificationHandler {...props} />);

describe('<NotificationHandler />', () => {
  test('should render properly', () => {
    wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });

  test('should render with notifications defined', () => {
    const spyDissmiss = jest.fn();
    wrapper = setup({ notifications: [{ id: 'ok' }], dismissNotification: spyDissmiss });
    expect(wrapper).toMatchSnapshot();
  });

  describe('dismissNotification', () => {
    test('it should set a timeout to dismiss the current notification if there is at least a notification', () => {
      const spyTimeOut = jest.spyOn(window, 'setTimeout').mockImplementation((fn) => fn());
      const spyDissmiss = jest.fn();
      wrapper = setup({ notifications: [{ id: 'ok' }], dismissNotification: spyDissmiss });

      expect(spyTimeOut).toHaveBeenCalledWith(spyDissmiss, 3500);
      expect(spyDissmiss).toHaveBeenCalled();
    });
  });

  describe('componentDidUpdate', () => {
    test('it should call dismissNotification if there is still a new notification to be displayed', () => {
      const spyTimeOut = jest.spyOn(window, 'setTimeout').mockImplementation((fn) => fn());
      const spyDissmiss = jest.fn();
      wrapper = setup({ notifications: [{ id: 'ok' }, { id: 'test' }], dismissNotification: spyDissmiss });
      wrapper.instance().componentDidUpdate({ notifications: [{ id: 'test' }], dismissSnackBar: spyDissmiss });

      expect(spyDissmiss).toHaveBeenCalledTimes(2);
    });
  });
});
