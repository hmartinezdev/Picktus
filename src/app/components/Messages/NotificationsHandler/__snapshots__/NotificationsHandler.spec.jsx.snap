// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotificationHandler /> should render properly 1`] = `
<div>
  <TransitionGroup
    appear={true}
    childFactory={[Function]}
    className="notifications"
    component="div"
  />
  <style
    jsx={true}
  >
    
          .notification__container {
            position: absolute;
            top: 4vh;
            right: -3px;
            transition: all 700ms ease-out;
            transform: translateX(120%);
            z-index: 1000;
          }

          .notification__container--entering {
            animation: slideIn 700ms ease-out;
          }

          .notification__container--entered {
            transform: translateX(0%);
          }

          .notification__container--exiting,
          .notification__container--exited {
            transform: translateX(120%);
          }

          @keyframes slideIn {
            from {
              transform: translateX(120%);
            }

            to {
              transform: translateX(0%);
            }
          }
        
  </style>
</div>
`;

exports[`<NotificationHandler /> should render with snackbars defined 1`] = `
<div>
  <TransitionGroup
    appear={true}
    childFactory={[Function]}
    className="notifications"
    component="div"
  >
    <Transition
      appear={false}
      enter={true}
      exit={true}
      in={false}
      key="ok"
      mountOnEnter={true}
      onEnter={[Function]}
      onEntered={[Function]}
      onEntering={[Function]}
      onExit={[Function]}
      onExited={[Function]}
      onExiting={[Function]}
      timeout={700}
      unmountOnExit={true}
    >
      <Component />
    </Transition>
  </TransitionGroup>
  <style
    jsx={true}
  >
    
          .notification__container {
            position: absolute;
            top: 4vh;
            right: -3px;
            transition: all 700ms ease-out;
            transform: translateX(120%);
            z-index: 1000;
          }

          .notification__container--entering {
            animation: slideIn 700ms ease-out;
          }

          .notification__container--entered {
            transform: translateX(0%);
          }

          .notification__container--exiting,
          .notification__container--exited {
            transform: translateX(120%);
          }

          @keyframes slideIn {
            from {
              transform: translateX(120%);
            }

            to {
              transform: translateX(0%);
            }
          }
        
  </style>
</div>
`;
